src/subscript/fmu_copy_revision/fmu_copy_revision.py

```diff
@@ -0,0 +1,677 @@
+        tdir = tempfile.TemporaryDirectory()
+        logger.info("The tmpdir is %s", tdir.name)
+        scriptname = join(tdir.name, "rsync.sh")
+        filterpatternname = join(tdir.name, "filterpattern.txt")
+        dirfilterpatternname = join(tdir.name, "dirfilterpattern.txt")
+
+        with open(scriptname, "w") as stream:
+            stream.write(SHELLSCRIPT)
+
+        with open(filterpatternname, "w") as stream:
+            stream.write(self.filter)
+
+        with open(dirfilterpatternname, "w") as stream:
+            stream.write(self.dirfilter)
+
+        logger.info("FILE FILTER FILE: %s", filterpatternname)
+        logger.info("DIR FILTER FILE: %s", dirfilterpatternname)
+        logger.debug("FILE FILTER IS\n: %s\n", self.filter)
+        logger.debug("DIR FILTER IS\n: %s\n", self.dirfilter)
+
+        self.nthreads = self.args.threads
+        if self.nthreads == 99:
+            # as default, leave one CPU free for other use
+            self.nthreads = cpu_count() - 1 if cpu_count() > 1 else 1
+
+        print(f"Doing copy using {self.nthreads} CPU threads, please wait...")
+
+        # the -R (--relative) is crucial for making filter profiles work!
+        rsyncargs = "-a -R --delete"
+
+        if self.args.dryrun:
+            rsyncargs += " --dry-run -v"
+
+        if self.args.verbosity:
+            rsyncargs += " -v"
+
+        # execute rsync, some explanations:
+        # filterpatternname: is the file to do per file filtering
+        # rsyncargs: see above
+        # self.keepfolders will be:
+        #   - 0 to delete all folders that are empty after copy/filtering
+        #   - 1 to keep all folders, even the empty ones
+        #   - 2 to keep all folders except those in dirfilterpatternname
+
+        command = [
+            "sh",
+            scriptname,
+            self.source,
+            self.target,
+            filterpatternname,
+            str(self.nthreads),
+            str(rsyncargs),
+            str(self.keepfolders),
+            dirfilterpatternname,
+        ]
+        logger.info(" ".join(command))
+
+        if sys.version_info[1] >= 7:
+            process = subprocess.run(
+                command, check=True, shell=False, capture_output=True
+            )
+        else:
+            # capture_output is not supported in 3.6 https://www.py4u.net/discuss/195326
+            process = subprocess.run(
+                command,
+                check=True,
+                shell=False,
+                stdout=subprocess.PIPE,
+                stderr=subprocess.PIPE,
+            )
+
+        if process.returncode != 0:
+            print(f"Process returncode: {process.returncode}")
+            print(process.stderr.decode())
+        stdout = process.stdout.decode().splitlines()
+
+        output = "\n".join(stdout[0:-2])
+        print(output)
+        timing = float(stdout[-1])
+        timing = time.strftime("%H hours %M minutes %S seconds", time.gmtime(timing))
```

Comment:
(this is nicely done with `humanize.precisedelta(sometimedelta)` but `humanize` is not in komodo)